crates/iota-keys/src/keystore.rs

```diff
@@ -175,9 +197,101 @@ pub struct Alias {
     pub public_key_base64: String,
 }
 
+#[expect(clippy::large_enum_variant)]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(
+    tag = "type",            // this makes {"type": "...", "value": …}
+    content = "value",       // name the payload field "value"
+    rename_all = "snake_case"
+)]
+pub enum StoredKey {
+    #[serde(with = "serde_iota_keypair")]
+    KeyPair(IotaKeyPair),
+    External {
+        source: String,
+        #[serde(
+            with = "serde_derivation_path::option",
+            default,
+            skip_serializing_if = "Option::is_none"
+        )]
+        derivation_path: Option<DerivationPath>,
+        public_key_base64: String,
```

Comment:
How about adding `with_flag` here? At least in your example the values contain the prefix byte for the type
And just to make sure, for IotaKeyPair you removed it, because we can just always compute it on the fly and therefore don't need it in the keystore file, right?